lib/_http_outgoing.js

```diff
@@ -517,11 +534,26 @@ OutgoingMessage.prototype.end = function(data, encoding, callback) {
     throw new TypeError('first argument must be a string or Buffer');
   }
 
+  var self = this;
+
   if (this.finished) {
+    if (data && data.length > 0) {
+      // Report that writing the data has failed
+      // because the stream was already 'ended'.
+      var err = new Error('write after end');
+      process.nextTick(function() {
+        self.emit('error', err);
+        if (callback) callback(err);
```

Comment:
I think this had been discussed some time back. See nodejs/node-v0.x-archive#7477 where this change originated from.